apps/api/v1/rest/courses/src/lib.ts

```diff
@@ -52,3 +54,44 @@ export function normalizeCourse(course: PrismaCourse): Course {
     terms: course.terms as unknown as string[],
   };
 }
+
+/**
+ * Constructs a Prisma query for the given filter parameters.
+ * @param parsedQuery The query object parsed by Zod.
+ */
+export function constructPrismaQuery(parsedQuery: Query): Prisma.CourseWhereInput {
+  const AND: Prisma.CourseWhereInput[] = [];
+
+  if (parsedQuery.department) AND.push({ department: parsedQuery.department });
```

Comment:
Personally, I just kinda decided that I like always using braces just so everything looks super consistent. Sometimes not including braces looks fine, but other times it doesn't and then I get confused what to look for lol. Not really a big deal, maybe something prettier can enforce? :shrug:

```suggestion
  if (parsedQuery.department) {
    AND.push({ department: parsedQuery.department })
  }
```

Reply:
Looks like there's an ESLint rule for that™: https://eslint.org/docs/latest/rules/curly#all

I can add it along with all of the other rules that we wanted to add before, since some of them require additional experimentation and are otherwise out of scope for this PR.